hazelcast/src/main/java/com/hazelcast/journal/EventJournalReader.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.journal;
+
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.ringbuffer.ReadResultSet;
+
+/**
+ * This interface provides methods to subscribe and read from an event journal.
+ *
+ * @param <E> journal event type
+ */
+public interface EventJournalReader<E> {
+
+    /**
+     * Subscribe to the event journal for this reader and a specific partition ID.
+     * The method will return the newest and oldest event journal sequence.
+     *
+     * @param partitionId the partition ID of the entries to which we are subscribing
+     * @return future with the initial subscriber state containing the newest and oldest event journal sequence
+     * @throws UnsupportedOperationException if the cluster version is lower than 3.9 or there is no event journal
+     *                                       configured for this map
+     * @since 3.9
+     */
+    ICompletableFuture<EventJournalInitialSubscriberState> subscribeToEventJournal(int partitionId);
+
+    /**
+     * Reads from the event journal. The returned future may throw {@link UnsupportedOperationException}
+     * if the cluster version is lower than 3.9 or there is no event journal configured for this map.
```

Comment:
let's not wait for another PR build cycle and merge as it is, I'll amend the javadoc with a direct commit